### PR TITLE
New Script: CUPS printing server

### PR DESCRIPTION
**Pull request by zdraganov**
## Description

Adding CUPS print server as LXE container installation

Fixes # (issue)

## Type of change
Please check the relevant option(s):

- [ ] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [X] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [ ] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [X] Testing performed (I have tested my changes, ensuring everything works as expected)
- [] Documentation updated (I have updated any relevant documentation)

### CODE REVIEW

File: install/cups-install.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+setting_up_container
+network_check
+update_os
+
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y curl
+$STD apt-get install -y sudo
+$STD apt-get install -y mc
+msg_ok "Installed Dependencies"
```
Comment (michelroegl-brunner) [inline install/cups-install.sh:21]:
```
$STD apt-get install -y \
  curl \
  sudo \
  mc 
```
This is Standard.

File: json/cups.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "CUPS",
+  "slug": "CUPS",
```
Comment (michelroegl-brunner) [inline json/cups.json:3]: `cups`

Lowercase

File: json/cups.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+              "cpu": 1,
+              "ram": 512,
+              "hdd": 8,
+              "os": "debian",
+              "version": "12"
+          }
+      }
+  ],
+  "default_credentials": {
+      "username": "root",
```
Comment (michelroegl-brunner) [inline json/cups.json:30]: Leave it as null if not needed.

File: install/cups-install.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+msg_info "Install CUPS"
```
Comment (michelroegl-brunner) [inline install/cups-install.sh:23]: Rename to Creating Service

File: install/cups-install.sh [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+$STD apt-get install -y cups
+msg_ok "Installed CUPS"
+
+msg_info "Add lpadmin to root group"
+$STD usermod -aG lpadmin root
+msg_ok "Added lpadmin to root group"
```
Comment (michelroegl-brunner) [inline install/cups-install.sh:29]: Put in same msg_info "Creating Service" as above

File: install/cups-install.sh [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+msg_info "Starting Service"
+systemctl enable -q --now cups
+msg_ok "Started Service"
```
Comment (michelroegl-brunner) [inline install/cups-install.sh:33]: Put in same msg_info "Creating Service" as above

File: install/cups-install.sh [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+msg_info "Allow remote administration"
+$STD cupsctl --remote-admin --remote-any --share-printers
+msg_ok "Allowed remote administration"
```
Comment (michelroegl-brunner) [inline install/cups-install.sh:37]: Put in same msg_info "Creating Service" as above

msg_ok "Configured Service"

**Review by michelroegl-brunner — CHANGES_REQUESTED** [review]

**Review by michelroegl-brunner — COMMENTED** [review]
Looks good for me, but @MickLesk or @havardthom still need to have a look to aprove the PR for merging.

**Comment by newzealandpaul** [comment]
@zdraganov, your JSON file is incorrect, the version of CUPS installed is this version: https://openprinting.github.io/cups/

Overall, does this need to be a script? It does not do a whole lot. I think using the Debian LXC script and simply doing an `apt install` is enough.

**Comment by zdraganov** [comment]
> Overall, does this need to be a script? It does not do a whole lot. I think using the Debian LXC script and simply doing an `apt install` is enough.

Most of the scripts are installing something, enable service - then configure it initially via the exposed web/terminal UI.
My feeling that cups is very common software that was just missing from the ProxmoxVE scripts and can help the people.

File: ct/cups.sh [hunk 8]
```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
+# Copyright (c) 2021-2024 tteck
```
Comment (MickLesk) [inline ct/cups.sh:3]: community-scripts ORG

File: ct/cups.sh [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+# Author: tteck (tteckster)
```
Comment (MickLesk) [inline ct/cups.sh:4]: YourUserName

File: ct/cups.sh [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+function update_script() {
+header_info
+check_container_storage
+check_container_resources
+if [[ ! -d /etc/cups ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+msg_info "Stopping ${APP} Service"
+systemctl stop cups
+msg_ok "Stopped ${APP} Service"
+
+msg_info "Updating..."
```
Comment (MickLesk) [inline ct/cups.sh:64]: msg_info "Updating ${APP}"

File: ct/cups.sh [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+apt install --only-upgrade cups &>/dev/null
+msg_ok "Update Successfully"
```
Comment (MickLesk) [inline ct/cups.sh:66]: msg_ok "Updated ${APP}"

File: install/cups-install.sh [hunk 12]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2021-2024 tteck
```
Comment (MickLesk) [inline install/cups-install.sh:3]: community-scripts ORG

File: install/cups-install.sh [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+# Author: tteck (tteckster)
```
Comment (MickLesk) [inline install/cups-install.sh:4]: Your UserName

File: install/cups-install.sh [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+  sudo \
+  mc
+msg_ok "Installed Dependencies"
+
+msg_info "Creating Service"
+$STD apt-get install -y cups
+$STD usermod -aG lpadmin root
+systemctl enable -q --now cups
+$STD cupsctl --remote-admin --remote-any --share-printers
+msg_ok "Configured Service"
```
Comment (MickLesk) [inline install/cups-install.sh:29]: Split this into 2 Sections

MSG Setup Cups

MSG Creating Service

**Comment by newzealandpaul** [comment]
> Most of the scripts are installing something, enable service - then configure it initially via the exposed web/terminal UI.
> My feeling that cups is very common software that was just missing from the ProxmoxVE scripts and can help the people.

But without hardware passthrough its not useful out of the box. 

Usually, but not always, installing involves more than a single command. 

@zdraganov I appreciate the effort you put into this. There were a significant number of errors, including misidentifying the package being installed. I don't believe we should merge this.

**Comment by zdraganov** [comment]
Makes sense. Closing this one.

**Comment by VNRARA** [comment]
Damn.. I was really looking for something like this.. I can't for the life of me get a LXC with CUPS going.. 
I just can't reach 631/admin whenever it's setup no matter the other settings I tweak.

**Pull request closed without merging**